react/components/molecules/search/search.js

```diff
@@ -0,0 +1,146 @@
+            onBlur: PropTypes.func,
+            onClear: PropTypes.func,
+            style: ViewPropTypes.style
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            value: undefined,
+            placeholder: "Search",
+            buttonProps: {},
+            onValue: () => {},
+            onFocus: () => {},
+            onBlur: () => {},
+            onClear: () => {},
+            style: {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            valueData: this.props.value,
+            buttonsVisible: true,
+            focused: false
+        };
+    }
+
+    focus = () => {
+        this.textInputComponent.focus();
+    };
+
+    blur = () => {
+        this.textInputComponent.blur();
+    };
+
+    clear = () => {
+        this.setState({ valueData: "" }, () => {
+            this.props.onValue("");
+            this.props.onClear();
+            this.focus();
+        });
+    };
+
+    _icon = () => {
+        return this.state.valueData ? "close" : "search";
+    };
+
+    onInputValue = value => {
+        this.setState({ valueData: value }, () => {
+            this.props.onValue(value);
+        });
+    };
+
+    onInputFocus = () => {
+        this.setState({ focused: true }, () => {
+            this.props.onFocus();
+        });
+    };
+
+    onInputBlur = () => {
+        this.setState({ focused: false }, () => {
+            this.props.onBlur();
+        });
+    };
+
+    onButtonPress = () => {
+        if (!this.state.valueData) return;
+        this.clear();
+    };
+
+    _style = () => {
+        return [styles.search, this.props.style];
+    };
+
+    render() {
+        return (
+            <View style={this._style()}>
+                <Input
+                    ref={el => (this.textInputComponent = el)}
+                    style={styles.input}
+                    value={this.state.valueData}
+                    placeholder={this.props.placeholder}
+                    placeholderTextColor={"#223645"}
+                    onValueUpdate={this.onInputValue}
+                    onFocus={this.onInputFocus}
+                    onBlur={this.onInputBlur}
+                />
+                <ButtonIcon
+                    style={styles.button}
+                    icon={this._icon()}
+                    size={24}
+                    iconStrokeColor={"#1d2631"}
+                    iconStrokeWidth={2}
+                    iconHeight={20}
+                    iconWidth={20}
+                    {...this.props.buttonProps}
+                    onPress={this.onButtonPress}
+                />
+            </View>
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    search: {
+        flexDirection: "row",
+        alignItems: "center",
+        position: "relative"
+    },
+    input: {
+        flex: 1,
+        color: "#223645",
+        paddingBottom: 0,
+        fontSize: 14,
+        fontFamily: "Roboto-Medium",

```

Comment:
Use baseStyle.FONT for now

Reply:
We can now use Roboto font as, both PRs with that font got merged

https://github.com/ripe-tech/ripe-id-mobile/pull/14
https://github.com/ripe-tech/ripe-components-react-native-storybook/pull/24

Reply:
I understand, but we will not use it like that, we use it as baseStyle.FONT. In the future, the FONT constant will be changed to point to the Roboto font, changing both ripe-id and ripe-robin apps